kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/eventually.kt

```diff
@@ -260,6 +266,21 @@ private class EventuallyControl(val config: EventuallyConfiguration) {
          appendLine(this.stackTraceToString())
       }
    }.toString()
+
+
+   fun Long.ensureInRange() : Long {
+      // Check if the duration is INFINITE which is default
+      if (config.duration == INFINITE) {
+         return INFINITE.inWholeMilliseconds
+      }
+
+      return when {
+         this in 0..INFINITE.inWholeMilliseconds -> this
+         else -> INFINITE.inWholeMilliseconds.also {
+            println("[WARN] end value $this is out of the valid range (0 to ${INFINITE.inWholeMilliseconds}), value is fixed to Duration.INFINITE.inWholeMilliseconds")
```

Comment:
```suggestion
            println("[WARN] end value $this is out of the valid range (0 to $it), value is fixed to Duration.INFINITE.inWholeMilliseconds")
```

Reply:
Should we not just error out and throw if the duration is invalid?

Reply:
think throwing error is more clear then changing value since people will not set duration close to inifnite

Reply:
And no one is going to spot the warn in the logs unless they look closely.
Failing the test makes more sense if its a bug, which having an incorrect duration is.